client/human/HumanClientFSM.cpp

```diff
@@ -815,6 +848,13 @@ boost::statechart::result WaitingForGameStart::react(const GameStart& msg) {
     GetGameRules().SetFromStrings(Client().GetGalaxySetupData().GetGameRules());
 
     bool is_new_game = !(loaded_game_data && ui_data_available);
+    Client().GetClientUI().PushWork(
```

Comment:
not clear about what's going on / why. `ProcessData` is called within `launch_async` but then it calls the `PushWork` function...  why in that order?